Cognite.Simulator.Utils/LicenseController.cs

```diff
@@ -0,0 +1,233 @@
+                    _logger.LogError(ex, "Error acquiring license");
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Releases the license after the timeout has elapsed. This is called by the timer.
+        /// </summary>
+        private void ReleaseLicenseAfterTimeout(object state)
+        {
+            lock (_lock)
+            {
+                _logger.LogInformation("Attempting to release license at scheduled time {ScheduledTime}", _scheduledReleaseTime.ToString("yyyy-MM-dd HH:mm:ss"));
+                if (_timeProvider.GetUtcNow() >= _scheduledReleaseTime)
+                {
+                    _releaseLicenseFunc();
+                    _licenseHeld = false;  
+                    _logger.LogInformation("License released successfully");
+                }
+                else
+                {
+                    _logger.LogInformation("License not released - current time {CurrentTime} is before scheduled release time", 
+                        _timeProvider.GetUtcNow().ToString("yyyy-MM-dd HH:mm:ss"));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Releases the license immediately. This is used externally to reset the internal state when an external process finds out
+        /// that the license is no longer held.
+        /// </summary>
+        public void ClearLicenseState() {
+            lock (_lock)
+            {
+                _licenseHeld = false;
+                _lastUsageTime = _timeProvider.GetUtcNow();
+                _scheduledReleaseTime = _timeProvider.GetUtcNow();
+                _releaseTimer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+                _logger.LogInformation("License released without callback");
+            }
+        }
+
+        private void PauseTimer() {
+            _logger.LogInformation("License usage started pausing release timer");
+            _releaseTimer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+        }
+
+        /// <summary>
+        /// Begins usage of the license.
+        /// </summary>
+        /// <returns>An <see cref="IDisposable"/> object that can be used to end the license usage.</returns>
+        public IDisposable BeginUsage()
+        {
+            lock (_lock)
+            {
+                _logger.LogInformation("Starting license usage");
+                PauseTimer();
+                return new LicenseUsageScope(this);
+            }
+        }
+
+        private void EndUsage()
+        {
+            lock (_lock)
+            {
+                _logger.LogInformation("Ending license usage, scheduling release");
+                _lastUsageTime = _timeProvider.GetUtcNow();
+                _scheduledReleaseTime = _lastUsageTime.Add(_licenseLockTime);
+                _logger.LogInformation("License release scheduled for {ReleaseTime}", 
+                _scheduledReleaseTime.ToString("yyyy-MM-dd HH:mm:ss"));
```

Comment:
```suggestion
```
this is noop?

Reply:
it should work: https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tostring?view=net-8.0

Reply:
ah, I misread it, sorry

Reply:
due to the formatting it looks like a separate code line! can you even read it?

Reply:
Is this formatting better? https://github.com/cognitedata/dotnet-simulator-utils/pull/239/commits/cba6a490d42c9183cf5bf86184ddfc1c3a6bc9d0